realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -301,7 +349,12 @@ public void execute(Realm realm) {
             @Override
             public void onSuccess() {
                 assertEquals(1, realm.where(AllTypes.class).count());
-                assertEquals(1, results.size());
+                // We cannot guarantee the async results get delivered from OS.
```

Comment:
@cmelchior I don't intend to be a nuisance but I'm pretty sure this callback isn't guaranteed to be delivered at this point exactly because of [the change in timing](https://github.com/realm/realm-java/issues/3427).

But previously it was possible to postpone change notifications until all pending queries were executed, maybe now it is not possible.

But if `isLoaded()` check needs to be added for async query results, then it is a breaking change.